services/navigation/navigation.go

```diff
@@ -81,21 +81,90 @@ type Properties struct {
 }
 
 // A Service controls the navigation for a robot.
+//
+// Mode example:
+//
+//	// Get the Mode the service is operating in
+//	mode, err := myNav.Mode(context.Background(), nil)
+//
+// SetMode example:
+//
+//	// Set the Mode the service is operating in to ModeWaypoint and begin navigation
+//	err := myNav.SetMode(context.Background(), navigation.ModeWaypoint, nil)
+//
+// Location example:
+//
+//	// Get the current location of the robot in the navigation service
+//	location, err := myNav.Location(context.Background(), nil)
+//
+// Waypoints example:
+//
+//	waypoints, err := myNav.Waypoints(context.Background(), nil)
+//
+// AddWaypoint example:
+//
+//	// Create a new waypoint with latitude and longitude values of 0 degrees
+//	// Assumes you have imported "github.com/kellydunn/golang-geo" as `geo`
+//	location := geo.NewPoint(0, 0)
+//
+//	// Add your waypoint to the service's data storage
+//	err := myNav.AddWaypoint(context.Background(), location, nil)
+//
+// RemoveWaypoint example:
+//
+//	// Create a new ObjectID
+//	// Assumes you have imported "go.mongodb.org/mongo-driver/bson/primitive"
+//	waypoint_id := primitive.NewObjectID()
+//
+//	// Remove the waypoint matching that ObjectID from the service's data storage
+//	err := myNav.RemoveWaypoint(context.Background(), waypoint_id, nil)
```

Comment:
```suggestion
//	// Assuming you have already called AddWaypoint once and the waypoint has not yet been reached
//       waypoints, err := myNav.Waypoints(context.Background(), nil)
//       if (err != nil || len(waypoints) == 0) {
//           return
//       }
//
//	// Remove the first waypoint from the service's data storage
//	err := myNav.RemoveWaypoint(context.Background(), waypoints[0].ID, nil)
```

Reply:
Creating a new object is aways going to fail. You need to provide the id of one of the outputs of Waypoints()

Reply:
great catch, thanks Nick :) !